README.md

```diff
@@ -3,4 +3,356 @@
+
+Finally, the dependencies can be installed into the newly created environment:
+
+```sh
+pip3 install -r requirements.txt
+```
+
+
+This will ensure your function has all dependencies it requires installed before you run it.  If you forgot to do this 
+before starting the Function locally you’ll receive an error reminding you to perform this step.
+
+### Run the Python Function Locally
+
+```sh
+sf run function start
+```
+
+This will start the function running locally on http://localhost:8080.  Messages logged by the running function will appear here.
+
+### Invoke the Running Python Function
+
+After starting the function:
+
+* open a new command line terminal 
+* navigate to the `hellopython` directory 
+* invoke the function by sending it a payload
+
+```sh
+sf run function --function-url http://localhost:8080 --payload '{}'
+```
+
+## Deploy the Python Function
+
+The following commands need to be executed from the **root of your SFDX project** so change into that directory now.
+
+### Commit your changes to Git
+
+All code changes made to a function will need to be staged and committed before you can deploy.
+
+```sh
+git add . 
+git commit -m "Trying out python functions"
+```
+
+Once everything is committed to git, run:
+
+```sh
+sf deploy functions --connected-org PythonScratch
+```
+
+This deployment process may take several minutes.
+
+### Invoke the Function from Apex
+
+The easiest way to invoke the function deployed to our scratch org is with some Apex code.  Generate an Apex class with:
+
+```sh
+sfdx force:apex:class:create \
+  --classname ApexTrigger \
+  --outputdir force-app/main/default/classes
+```
+
+Open `force-app/main/default/classes/ApexTrigger.cls` and replace it with the following code:
+
+```java
+public with sharing class ApexTrigger {
+    public static void runFunction() {
+        System.debug('Running hellopython');
+        functions.Function fn = functions.Function.get('PythonFunctionsAlpha.hellopython');
+        functions.FunctionInvocation invocation = fn.invoke('{}');
+        System.debug('Response: ' + invocation.getResponse());
+    }
+}
+```
+
+This code will:
+
+* Lookup the reference to our function using the `functions.Function.get` method 
+* Invoke the function with an empty json payload
+* Print the response
+
+Upload this Apex class to your scratch org with
+
+```sh
+sfdx force:source:push --targetusername PythonScratch
+```
+
+Open a developer console:
+
+```sh
+sfdx force:org:open -p /_ui/common/apex/debug/ApexCSIPage
+```
+
+Then execute the function.
+
+On a **macOS / Linux system** you can execute the function with:
+
+```sh
+echo "ApexTrigger.runFunction();" | sfdx force:apex:execute -f /dev/stdin
+```
+
+On a **Microsoft Windows system** you can execute the function with:
+
+```sh
+echo "FunctionApex.test();" | sfdx force:apex:execute
+```
+
+The developer console will show a log entry after the function executes which you can double-click to open.  Toggle the 
```

Comment:
It wasn't clear to me where I should be looking for this log entry. I was expecting something to appear in the main (middle) panel, when in fact the thing that appeared was in the bottom panel, and I had to double click that to get the middle panel to actually show anything. Whilst the current wording does mostly say that, this was still a bit counter-intuitive. 

Is this something people familiar with Salesforce will already know how to use, or do we need to tweak the wording here to make this even more obvious (eg mention "show a log entry in the bottom panel").

Reply:
Addressed in #34.